js/state/siteUtil.js

```diff
@@ -600,14 +600,13 @@ module.exports.isImportedBookmark = function (siteDetail) {
  */
 module.exports.isHistoryEntry = function (siteDetail) {
   if (siteDetail && typeof siteDetail.get('location') === 'string') {
-    if (siteDetail.get('location').startsWith('about:')) {
+    const tags = siteDetail.get('tags')
+    if (siteDetail.get('location').startsWith('about:') ||
+      module.exports.isDefaultEntry(siteDetail) ||
+      isBookmarkFolder(tags)) {
       return false
     }
-    if (module.exports.isDefaultEntry(siteDetail)) {
-      // This is a Brave default newtab site
-      return false
-    }
-    return !!siteDetail.get('lastAccessedTime') && !isBookmarkFolder(siteDetail.get('tags'))
+    return !!siteDetail.get('lastAccessedTime') || !tags || tags.size === 0
```

Comment:
i don't think there'd be a crash any more due to the change to downgrade the `throw` into `console.log`.

On bookmarks == history, I think that's not obvious to a user; so if I synced my bookmarks I'd expect history to be empty. It would only make sense if I understood the sites data model. However, this is how it currently happens in `master` so this is probably ok.